plugins/source/aws/client/client.go

```diff
@@ -368,25 +368,24 @@ func Configure(ctx context.Context, logger zerolog.Logger, spec specs.Source) (s
 			return nil, err
 		}
 
-		// This is a work-around to skip disabled regions
-		// https://github.com/aws/aws-sdk-go-v2/issues/1068
-		res, err := ec2.NewFromConfig(awsCfg).DescribeRegions(ctx,
-			&ec2.DescribeRegionsInput{AllRegions: aws.Bool(false)},
-			func(o *ec2.Options) {
-				o.Region = defaultRegion
-				if account.DefaultRegion != "" {
-					o.Region = account.DefaultRegion
-				}
+		// By default we should use the default region (us-east-1)
+		regionsToCheck := []string{defaultRegion}
+		// If user specifies a Default Region we should use it
+		if account.DefaultRegion != "" {
+			regionsToCheck = []string{account.DefaultRegion}
+			// If no default region and * is not specified we should use all specified regions
+		} else if len(localRegions) > 0 && !isAllRegions(localRegions) {
+			regionsToCheck = localRegions
+		}
 
-				if len(localRegions) > 0 && !isAllRegions(localRegions) {
-					o.Region = localRegions[0]
-				}
-			})
-		if err != nil {
-			logger.Warn().Str("account", account.AccountName).Err(err).Msg("Failed to find disabled regions for account")
-			continue
+		for _, region := range regionsToCheck {
+			enabledRegions, err := getEnabledRegions(ctx, awsCfg, region)
+			if err != nil {
+				logger.Warn().Str("account", account.AccountName).Err(err).Msgf("Failed to find disabled regions for account when checking: %s", region)
+				continue
+			}
+			account.Regions = filterDisabledRegions(localRegions, enabledRegions)
```

Comment:
Should we maybe break out of the loop here? Otherwise we'll do it for all regions even though only the last (working) result will count

Reply:
Fixed